tests/mockdata/xgm.py

```diff
@@ -56,3 +56,40 @@ class XGMReduced(XGMD):
         ("classId", None, "DoocsXGMReduced"),
         ("location", None, "XGM.3356.SQS")
     ]
+
+class XGMWithData(DeviceBase):
```

Comment:
I didn't compare the exact keys, but would it make sense to inherit one of the above to save some definitions, and just overwrite `write_instrument()`?

Reply:
Indeed, except for the shape of the instrument array, which I tried to keep as 1D to avoid extra complications, but I can just override that in the daughter class.